File: nahomy-c2-backend/src/domain/entities/EjecucionAutomatizada.ts
export type EstadoEjecucion = 'Pendiente' | 'En Progreso' | 'Exitoso' | 'Fallido'

export class EjecucionAutomatizada {
  constructor(
    private readonly _id: number,
    private readonly _procesoId: number,
    private readonly _funcionId: number,
    private _ordenEjecucion: number,
    private _resultadoRaw: string | null,
    private _estado: EstadoEjecucion,
  ) {}

  get id(): number {
    return this._id
  }

  get procesoId(): number {
    return this._procesoId
  }

  get funcionId(): number {
    return this._funcionId
  }

  get ordenEjecucion(): number {
    return this._ordenEjecucion
  }

  get resultadoRaw(): string | null {
    return this._resultadoRaw
  }

  get estado(): EstadoEjecucion {
    return this._estado
  }

  ejecutar(resultado: string): void {
    this._resultadoRaw = resultado
    this._estado = 'Exitoso'
  }

  marcarFallido(error: string): void {
    this._resultadoRaw = error
    this._estado = 'Fallido'
  }

  reordenar(nuevoOrden: number): void {
    this._ordenEjecucion = nuevoOrden
  }
}

File: packages/server/src/domain/entities/ejecucion_automatizada.py
EstadoEjecucion = str


class EjecucionAutomatizada:
    def __init__(
        self,
        id: int,
        proceso_id: int,
        funcion_id: int,
        orden_ejecucion: int,
        resultado_raw: str | None,
        estado: EstadoEjecucion,
    ):
        self._id = id
        self._proceso_id = proceso_id
        self._funcion_id = funcion_id
        self._orden_ejecucion = orden_ejecucion
        self._resultado_raw = resultado_raw
        self._estado = estado

    @property
    def id(self) -> int:
        return self._id

    @property
    def proceso_id(self) -> int:
        return self._proceso_id

    @property
    def funcion_id(self) -> int:
        return self._funcion_id

    @property
    def orden_ejecucion(self) -> int:
        return self._orden_ejecucion

    @property
    def resultado_raw(self) -> str | None:
        return self._resultado_raw

    @property
    def estado(self) -> EstadoEjecucion:
        return self._estado

    def ejecutar(self, resultado: str) -> None:
        self._resultado_raw = resultado
        self._estado = "Exitoso"

    def marcar_fallido(self, error: str) -> None:
        self._resultado_raw = error
        self._estado = "Fallido"

    def reordenar(self, nuevo_orden: int) -> None:
        self._orden_ejecucion = nuevo_orden
